.github/workflows/tests.yml

```diff
@@ -0,0 +1,56 @@
+name: Tests
+on:
+  push:
+    branches:
+      - '**'
+    tags-ignore:
+      - '**'
+  # Triggers the workflow on labeled PRs only.
+  pull_request_target:
+    types: [ labeled ]
+# Ensures that only the latest commit is running for each PR at a time.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.ref }}
+  cancel-in-progress: true
+jobs:
+  Tests:
+    if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
+    name: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu, windows, macOS ]
+        include:
+          - os: windows
+            gradlewSuffix: .bat
+    runs-on: ${{ matrix.os }}-latest
+    steps:
+      - name: Install Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.20.x
+          cache: false
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+      - name: Setup Artifactory
+        run: |
+          go install github.com/jfrog/jfrog-testing-infra/local-rt-setup@latest
+          ~/go/bin/local-rt-setup
+        env:
+          RTLIC: ${{secrets.RTLIC}}
+          GOPROXY: direct
+
+      - name: Install Java
+        uses: actions/setup-java@v3
+        with:
+          java-version: "8"
+          distribution: "temurin"
+
+      - name: Run tests
+        run: ./gradlew${{ matrix.gradlewSuffix }} clean test
+        env:
+          CLIENTTESTS_ARTIFACTORY_URL: http://localhost:8081/artifactory
+          CLIENTTESTS_ARTIFACTORY_USERNAME: admin
+          CLIENTTESTS_ARTIFACTORY_PASSWORD: password
```

Comment:
These credentials should be moved to the tests boot start and used in the absence of an Env.